src/all/namicomi/src/eu/kanade/tachiyomi/extension/all/namicomi/Namicomi.kt

```diff
@@ -0,0 +1,386 @@
+
+    override fun popularMangaRequest(page: Int): Request =
+        sortedMangaRequest(page, "views")
+
+    override fun popularMangaParse(response: Response): MangasPage =
+        mangaListParse(response)
+
+    // Latest manga section
+
+    override fun latestUpdatesRequest(page: Int): Request =
+        sortedMangaRequest(page, "publishedAt")
+
+    override fun latestUpdatesParse(response: Response): MangasPage =
+        mangaListParse(response)
+
+    private fun mangaListParse(response: Response): MangasPage {
+        if (response.code == 204) {
+            return MangasPage(emptyList(), false)
+        }
+
+        val mangaListDto = response.parseAs<MangaListDto>()
+        val coverSuffix = preferences.coverQuality
+
+        val mangaList = mangaListDto.data.map { mangaDataDto ->
+            val fileName = mangaDataDto.relationships
+                .firstInstanceOrNull<CoverArtDto>()
+                ?.attributes?.fileName
+
+            helper.createBasicManga(mangaDataDto, fileName, coverSuffix, extLang)
+        }
+
+        return MangasPage(mangaList, mangaListDto.hasNextPage)
+    }
+
+    // Search manga section
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.startsWith(NamicomiConstants.prefixIdSearch)) {
+            val mangaId = query.removePrefix(NamicomiConstants.prefixIdSearch)
+
+            if (!helper.containsId(mangaId)) {
+                throw Exception(helper.intl["invalid_manga_id"])
+            }
+
+            // If the query is an ID, return the manga directly
+            val url = NamicomiConstants.apiSearchUrl.toHttpUrl().newBuilder()
+                .addQueryParameter("ids[]", query.removePrefix(NamicomiConstants.prefixIdSearch))
+                .addQueryParameter("includes[]", NamicomiConstants.coverArt)
+                .build()
+
+            return GET(url, headers, CacheControl.FORCE_NETWORK)
+        }
+
+        val tempUrl = NamicomiConstants.apiSearchUrl.toHttpUrl().newBuilder()
+            .addQueryParameter("limit", NamicomiConstants.mangaLimit.toString())
+            .addQueryParameter("offset", helper.getMangaListOffset(page))
+            .addQueryParameter("includes[]", NamicomiConstants.coverArt)
+
+        val actualQuery = query.replace(NamicomiConstants.whitespaceRegex, " ")
+        if (actualQuery.isNotBlank()) {
+            tempUrl.addQueryParameter("title", actualQuery)
+        }
+
+        val finalUrl = helper.filters.addFiltersToUrl(
+            url = tempUrl,
+            filters = filters.ifEmpty { getFilterList() },
+            extLang = extLang,
+        )
+
+        return GET(finalUrl, headers, CacheControl.FORCE_NETWORK)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)
+
+    // Manga Details section
+
+    override fun getMangaUrl(manga: SManga): String =
+        "$baseUrl/en" + manga.url + "/${helper.titleToSlug(manga.title)}"
```

Comment:
always "en" in url?

Reply:
Changed it to insert the extension language instead.
NamiComi doesn't support UI translations in all available languages, but the website will automatically redirect to the English version in that case.